src/Agent.Worker/ExecutionContext.cs

```diff
@@ -515,6 +521,16 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
                 }
             }
 
+            try
+            {
+                HostContext.SecretMasker.MinSecretLength = AgentKnobs.MaskedSecretMinLength.GetValue(this).AsInt();
+            }
+            catch (ArgumentException ex)
```

Comment:
>ArgumentException [](http://example.com/codeflow?start=19&length=17)

[suggestion/not blocking]

It's better not to use an exception handler here -- instead avoid the error by checking the impact

A few problems with this approach:
1. ArgumentException is fairly broad and it may end up catching an ArgumentException you weren't expecting and cause unexpected behavior
2. it raises a 1st change exception which causes more noise in the debugger

Reply:
Resolved in separate PR - https://github.com/microsoft/azure-pipelines-agent/pull/4137